templates/buy-me-a-coffee-app/src/components/Header.tsx

```diff
@@ -0,0 +1,77 @@
+import { useState, ReactNode, useEffect } from 'react';
+import { Flex, IconButton, Theme, Tooltip } from '@radix-ui/themes';
+import styles from './Header.module.css';
+import { ThemeToggle } from './ThemeToggle';
+import { GitHubLogoIcon } from '@radix-ui/react-icons';
+import NextLink from 'next/link';
+import { classNames } from '../utils/classNames';
+import { Logo } from './Logo';
+import { ConnectButton } from '@rainbow-me/rainbowkit';
+
+export type HeaderProps = {
+  children?: ReactNode;
+  gitHubLink?: string;
+  ghost?: boolean;
+};
+
+type ScrollState = 'at-top' | 'scrolling-up' | 'scrolling-down';
+
+export function Header({ children, gitHubLink, ghost }: HeaderProps) {
+  const [scrollState, setScrollState] = useState<ScrollState>('at-top');
+
+  useEffect(() => {
+    let previousScrollY = window.scrollY;
+
+    const handleScroll = () => {
+      const direction = previousScrollY < window.scrollY ? 'scrolling-down' : 'scrolling-up';
+      const state = window.scrollY < 30 ? 'at-top' : direction;
+      previousScrollY = window.scrollY;
+      setScrollState(state);
+    };
+
+    if (ghost) {
+      addEventListener('scroll', handleScroll, { passive: true });
+    } else {
+      removeEventListener('scroll', handleScroll);
+    }
+
+    handleScroll();
+    return () => removeEventListener('scroll', handleScroll);
+  }, [ghost]);
+
+  //
+  return (
+    <Theme asChild className="radix-themes-custom-fonts">
+      <div
+        data-scroll-state={scrollState}
+        className={classNames(styles.HeaderRoot, ghost ? styles.ghost : '')}
+      >
+        <div className={styles.HeaderInner}>
+          <Flex align="center" position="absolute" top="0" bottom="0" left="0" pl="4">
+            <NextLink href="/" passHref legacyBehavior>
+              <Logo />
+            </NextLink>
+          </Flex>
+
+          <Flex align="center" gap="5" position="absolute" top="0" bottom="0" right="0" pr="4">
+            {children}
+
+            <ConnectButton />
```

Comment:
We should probably get in the habit to have some sort of standard comments in the TS and in the HTML for find critical Onchain parts.

This can be super dope when we have articles that explain how to work with this App and the comments can be use as coordinate across the App.